cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaSpecChecker.java

```diff
@@ -25,31 +26,37 @@
  * lead to problems.
  */
 public class KafkaSpecChecker {
-
-    private KafkaSpec spec;
-    private KafkaCluster kafkaCluster;
-    private ZookeeperCluster zkCluster;
+    private final KafkaCluster kafkaCluster;
+    private final ZookeeperCluster zkCluster;
+    private final String kafkaBrokerVersion;
 
     private final static Pattern VERSION_REGEX = Pattern.compile("(\\d\\.\\d+).*");
```

Comment:
I know it's not part of this PR, but this regex looks dodgy to me. I would expect it to be `\d+(\.\d+)+.*` (not quoted for use in a String literal) (see kafka.api.ApiVersion#apply for how they're actually validated).

Reply:
@tombentley `kafka.api.ApiVersion#apply` seems to take the first and the second segments. But I think it does something different then we do. We do not validate the protocol / format version here. We want to to get the numeric part only and compare it against the version if they match which tells us they are from the same version. The actual version is validated in other place in our case.

So I think the original pattern works well apart from not expecting any double-digit major Kafka versions. But the use of the matcher seems to be weird. The group 0 returns the whole pattern. So that seems to be wrong because when you use protocol format as `2.6-IV0` it would never match `2.6.0`. So I tried to fix that by using the right group.

Please have a look and let me know if I misunderstood what you meant.

Reply:
Group 0 is always the whole match, and the capturing groups are then 1-based. You're right about the first two numbers thing, so `(\d+\.\d+).*` should be enough, that way the `.*` can match patch versions (and the internal version part, which I'd never expect to be present in practice). Probably a good idea to add a comment too.